### PR TITLE
[feature/version-hide] Add support for version.hide server setting

### PR DESCRIPTION
**Pull request by felix-schwarz**
## Description
Adds support for `version.hide` server setting.

## Related Issue
#426

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: ownCloud/Bookmarks/BookmarkViewController.swift [hunk 1]
```diff
@@ -536,7 +536,26 @@ class BookmarkViewController: StaticTableViewController {
 					} else {
 						OnMainThread {
 							hudCompletion({
-								weakSelf.presentingViewController?.dismiss(animated: true, completion: nil)
+								if issue != nil {
+									self?.bookmark?.authenticationData = nil
+
+									let issuesViewController = ConnectionIssueViewController(displayIssues: issue?.prepareForDisplay(), completion: { [weak self] (response) in
```
Comment (hosy) [inline ownCloud/Bookmarks/BookmarkViewController.swift]: `[weak self]` is here not needed, because you already using `weakself` from above

**Review by hosy — CHANGES_REQUESTED** [review]

File: ownCloud/Bookmarks/BookmarkViewController.swift (hunk 1 above)
Comment (felix-schwarz) [reply ownCloud/Bookmarks/BookmarkViewController.swift]: That `[weak self]` is to make the reference to `self` weak in the `ConnectionIssueViewController` completion block. `weakSelf` above is actually a strong reference to self derived from a previous `[weak self]`. I'll change the naming to make this less confusing.

Comment (felix-schwarz) [reply ownCloud/Bookmarks/BookmarkViewController.swift]: Changed the naming in the latest commit.

**Review by hosy — APPROVED** [review]

**Comment by jesmrec** [comment]
Approved, now accounts are acceded even when the version string in `status.php` is hidden

**Review by jesmrec — APPROVED** [review]

**Pull request merged**